src/interchange/src/confluent.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::{bail, Result};
+use byteorder::{BigEndian, ByteOrder};
+
+/// Extracts the schema_id placed in front of the serialized message by the confluent stack
+/// Optionally expect an empty
+///
+/// This function returns the schema_id and a sublice of the rest of the buffer
+pub fn extract_schema_id(buf: &[u8], expect_msg_id: bool) -> Result<(i32, &[u8])> {
+    // The first byte is a magic byte (0) that indicates the Confluent
+    // serialization format version, and the next four bytes are a big
+    // endian 32-bit schema ID.
+    //
+    // https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html#wire-format
+    //
+    // For formats like protobuf, confluent adds additional information related to
+    // which message in the proto file
+
+    let expected_len = 5 + expect_msg_id as usize;
+
+    if buf.len() < expected_len {
+        bail!(
+            "Confluent-style schema datum is too few bytes: expected at least {} bytes, got {}",
+            expected_len,
+            buf.len()
+        );
+    }
+    let magic = buf[0];
+    let schema_id = BigEndian::read_i32(&buf[1..5]);
+
+    if magic != 0 {
+        bail!(
+            "wrong Confluent-style schema serialization magic: expected 0, got {}",
+            magic
+        );
+    }
+
+    if expect_msg_id {
+        let message_id = buf[5];
+        if message_id != 0 {
+            bail!(
+                "wrong Confluent-style message id: only 0 is supported, but found: {}",
```

Comment:
I would consider rewording this to make it more clear that this is a temporary limitation, and possibly creating a github issue for people to comment on for full support:

```suggestion
                "Unsupported Confluent-style protobuf message descriptor id: expected 0 but found: {}. \
                 See https://github.com/MaterializeInc/materialize/issues/####",
```

Otherwise, a slightly better design for this function (that leads to worse error messages, but keeps the existing one before my comment) would be for it to take an `Option<u8>` so that it doesn't hard-code anything about the valid values, which should in principle be a matter for some other piece of code.

Reply:
I think better error messages is worth the worse api, will change the wording tho!